mllib/src/main/scala/org/apache/spark/ml/evaluation/ClusteringEvaluator.scala

```diff
@@ -107,15 +106,18 @@ class ClusteringEvaluator @Since("2.3.0") (@Since("2.3.0") override val uid: Str
 
   @Since("2.3.0")
   override def evaluate(dataset: Dataset[_]): Double = {
-    SchemaUtils.checkColumnType(dataset.schema, $(featuresCol), new VectorUDT)
+    SchemaUtils.validateVectorCompatibleColumn(dataset.schema, $(featuresCol))
     SchemaUtils.checkNumericType(dataset.schema, $(predictionCol))
 
+    val vectorCol = DatasetUtils.columnToVector(dataset, $(featuresCol))
+    val df = dataset.select(col($(predictionCol)),
```

Comment:
not sure this is the right way. Probably we can face the same issue everywhere we are using `DatasetUtils.columnToVector`. Probably it is better to fix the problem there.

Reply:
@mgaido91  Thanks for your reviewing!
I have considered this, however there exists a problem:
if we want to append metadata into the transformed column (like using method `.as(alias: String, metadata: Metadata)`) in `DatasetUtils.columnToVector`, how can we get the name of transformed column?
The only way to do this I know is:
```
val metadata = ...
val vectorCol = ..
val vectorName = dataset.select(vectorCol) .schema.head.name
vectorCol.as(vectorName, metadata)
```

Reply:
we have the new column we are returning, so we can easily get its name with `.name`

Reply:
@mgaido91  I think it maybe nice to first add a name getter for column

Reply:
we can propose that